### PR TITLE
Fix issues with plugin app_config field.

### PR DESCRIPTION
**Pull request by cyberstar**
Introduces:

* Perform app_config filtering for plugins;
* display app configs that exist but either not app hooked or page is not published in a help text;
* prevent 500 if app hook was deleted (for plugins);
* change help text for app_config field (no migration);
* test cases for testing if error message is displayed and if page with plugins are accessible even though app hook or JobsConfig was deleted.

### CODE REVIEW

**Comment by cyberstar** [comment]
rebased.

File: aldryn_jobs/boilerplates/bootstrap3/templates/aldryn_jobs/plugins/categories_list.html [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {% load i18n apphooks_config_tags %}
  <div class="aldryn aldryn-jobs aldryn-jobs-categories">
+{% if plugin_configuration_error %}
```
Comment (FinalAngel) [inline aldryn_jobs/boilerplates/bootstrap3/templates/aldryn_jobs/plugins/categories_list.html]: @cyberstar be aware that this changed in: https://github.com/aldryn/aldryn-jobs/pull/61

File: aldryn_jobs/boilerplates/bootstrap3/templates/aldryn_jobs/plugins/latest_entries.html [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
 <div class="plugin plugin-jobs">
-    <p>{% blocktrans with total=instance.job_offers.count %}Showing {{ total }} Job Offers{% endblocktrans %}
-	<ul class="jobs-latest-entries">
-	{% for job in instance.job_offers %}
-		<li><p><a href="{{ job.get_absolute_url }}">{{ job }}</a></p></li>
-	{% empty %}
-		<li class="jobs-empty"><p>{% trans "There are currently no job openings." %}</p></li>
-	{% endfor %}
-	</ul>
+{% if plugin_configuration_error %}
```
Comment (FinalAngel) [inline aldryn_jobs/boilerplates/bootstrap3/templates/aldryn_jobs/plugins/latest_entries.html]: @cyberstar be aware that this changed in: https://github.com/aldryn/aldryn-jobs/pull/61

File: aldryn_jobs/boilerplates/legacy/templates/aldryn_jobs/plugins/categories_list.html [hunk 3]
```diff
@@ -1,5 +1,10 @@
 {% load i18n apphooks_config_tags %}
  <div class="aldryn aldryn-jobs aldryn-jobs-categories">
+{% if plugin_configuration_error %}
```
Comment (FinalAngel) [inline aldryn_jobs/boilerplates/legacy/templates/aldryn_jobs/plugins/categories_list.html]: @cyberstar be aware that this changed in: https://github.com/aldryn/aldryn-jobs/pull/61

**Comment by cyberstar** [comment]
rebased.

File: aldryn_jobs/boilerplates/bootstrap3/templates/aldryn_jobs/plugins/categories_list.html [hunk 4]
```diff
@@ -2,6 +2,11 @@
 
 <div class="aldryn aldryn-jobs aldryn-jobs-categories">
     <div class="list-group">
+    {% if plugin_configuration_error %}
```
Comment (FinalAngel) [inline aldryn_jobs/boilerplates/bootstrap3/templates/aldryn_jobs/plugins/categories_list.html]: missing indent

File: aldryn_jobs/boilerplates/bootstrap3/templates/aldryn_jobs/plugins/categories_list.html [hunk 5]
```diff
@@ -2,6 +2,11 @@
...
+        {% if request.user.is_staff or request.user.is_superuser %}
+            <p class="config-errors">{{ plugin_configuration_error }}</p>
```
Comment (FinalAngel) [inline aldryn_jobs/boilerplates/bootstrap3/templates/aldryn_jobs/plugins/categories_list.html]: use ``<div class="alert alert-danger">``

File: aldryn_jobs/boilerplates/bootstrap3/templates/aldryn_jobs/plugins/categories_list.html (hunk 4 above)
Comment (FinalAngel) [inline aldryn_jobs/boilerplates/bootstrap3/templates/aldryn_jobs/plugins/categories_list.html]: this needs to be outside of the list group

File: aldryn_jobs/cms_plugins.py [hunk 6]
```diff
@@ -6,19 +6,43 @@
...
+                     else '')
+
+        if not namespace_is_apphooked(namespace):
+            context['plugin_configuration_error'] = _(
+                'There is an error in plugin configuration: selected Job '
+                'config is not available. Please switch to edit mode and '
+                'change plugin app_config settings to use valid config. '
+                'Also note that aldryn-jobs should be used at least once '
+                'as an apphook for that config.')
+
```
Comment (mkoistinen) [inline aldryn_jobs/cms_plugins.py]: This might be best quoted using triple-quotes """...""" given its length.

**Comment by mkoistinen** [comment]
LGTM

**Comment by FinalAngel** [comment]
LGTM

**Comment by cyberstar** [comment]
rebased

**Comment by FinalAngel** [comment]
FE LGTM

**Pull request merged**